code/modules/mob/living/carbon/carbon.dm

```diff
@@ -496,6 +496,10 @@ var/const/GALOSHES_DONT_HELP = 8
 	if( (!mask_only && head && (head.flags & HEADCOVERSMOUTH)) || (!head_only && wear_mask && (wear_mask.flags & MASKCOVERSMOUTH)) )
 		return 1
 
+/mob/living/carbon/proc/cannot_facehug(head_only = 0, mask_only = 0)
+	if( (!mask_only && head && (head.flags & CANNOTFACEHUG)) || (!head_only && wear_mask && (wear_mask.flags & CANNOTFACEHUG)) )
+		return 1
+

```

Comment:
Since you're using a whole new flag I don't think it's necessary to make a proc with two arguments to only check the headgear or mask. Simply using `if( head && (head.flags & CANNOTFACEHUG))` should be enough since only headgears can have that flag.